src/Interfaces/AbstractPolyhedron_functions.jl

```diff
@@ -483,6 +507,28 @@ function _linear_map_hrep_helper(M::AbstractMatrix{N}, P::AbstractPolyhedron{N},
     return constraints_MP
 end
 
+# preconditions have been checked in the caller function
+function _linear_map_hrep_lift(M::AbstractMatrix{N}, P::AbstractPolyhedron{N}) where {N<:Real}
+    m, n = size(M)
+
+    # we extend M to an invertible m x m matrix by appending m-n columns
+    # orthogonal to the column space of M
+    Mext, inv_Mext = extend(M, check_rank=false)
+
+    # append zeros to the existing constraints, in the last m-n coordinates
+    cext = [HalfSpace(vcat(c.a, zeros(N, m-n)), c.b) for c in constraints_list(P)]
+
+    # now fix the last m-n coordinates to zero
+    id_out = Matrix(one(N)*I, m-n, m-n)
+    cext = vcat(cext, [HalfSpace(vcat(zeros(N, n), id_out[i, :]), zero(N)) for i in 1:(m-n)],
```

Comment:
Is `vcat(zeros(N, n), id_out[i, :])` not just `SingleEntryVector(n+i, m, one(N))`?

Reply:
That is correct, but note that `c.a` is not in general a `SingleEntryVector`. However, we are allowed to make the change you propose because `HPolytope` does not still carry the array type information of it's constraints list. But that is planned  (https://github.com/JuliaReach/LazySets.jl/issues/1802 and related issues) and then using mixed types is not a good idea (or this function will simply break). I'm not sure what to do..

Reply:
I propose to leave it like it is now and open an issue for later with reference to [this idea](https://github.com/JuliaReach/LazySets.jl/pull/1939/files#r375331312) as well.

Reply:
I do not think that a concrete `Vector` (that you use here) is always correct for the same reason. So I suggest to change it :smile:

Reply:
`vcat` does the conversion to a common type for you:
```julia
julia> v = [1, 2]
2-element Array{Int64,1}:
 1
 2

julia> vcat(v, [3.0, 4.0])
4-element Array{Float64,1}:
 1.0
 2.0
 3.0
 4.0
```
So I think we are good anyway.

Reply:
True, but it is not recursive,

```julia
julia> x = SingleEntryVector(1, 2, 1.0)
2-element SingleEntryVector{Float64}:
 1.0
 0.0

julia> y = rand(2)
2-element Array{Float64,1}:
 0.11055345051539978
 0.8948212121880166 

julia> vcat([HalfSpace(x, 1.0)], [HalfSpace(y, 2.0)])
2-element Array{HalfSpace{Float64,VN} where VN<:AbstractArray{Float64,1},1}:
 HalfSpace{Float64,SingleEntryVector{Float64}}([1.0, 0.0], 1.0)                     
 HalfSpace{Float64,Array{Float64,1}}([0.11055345051539978, 0.8948212121880166], 2.0)
```

Reply:
Okay, as you prefer.